### PR TITLE
Fix _execute_and_process_stdout.

### PR DESCRIPTION
**Pull request by winterfroststrom**
Changes _execute_and_process_stdout to match SnippetClient._read_protocol_line.

This should fix an issue with _execute_and_process_stdout where the last few lines of output were missing. This behaviour is actually implied by the test test_execute_and_process_stdout_when_cmd_exits, which is changed.

---
This change is [Reviewable](https://reviewable.io/reviews/google/mobly/456)

### CODE REVIEW

**Comment by xpconanfan** [comment]
Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---

*[mobly/controllers/android_device_lib/adb.py, line 208 at r1](https://reviewable.io/reviews/google/mobly/456#-LEVREEG3xFik2sES1m-:-LEVREEG3xFik2sES1m0:byf5ovv) ([raw file](https://github.com/google/mobly/blob/69c602b0ca226dfc73c16408380b696eb19896ca/mobly/controllers/android_device_lib/adb.py#L208)):*
> ```Python
>         out = '[elided, processed via handler]'
>         try:
>             while True:
> ```

this is the bug?...

---


*Comments from [Reviewable](https://reviewable.io/reviews/google/mobly/456)*

**Comment by winterfroststrom** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[mobly/controllers/android_device_lib/adb.py, line 208 at r1](https://reviewable.io/reviews/google/mobly/456#-LEVREEG3xFik2sES1m-:-LEVUNcTBe5juWPNezDG:b-ctfvrk) ([raw file](https://github.com/google/mobly/blob/69c602b0ca226dfc73c16408380b696eb19896ca/mobly/controllers/android_device_lib/adb.py#L208)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

this is the bug?...
</blockquote></details>

Yes.

Basically, if the process dies, then apparently stdout can still have buffered output, which doesn't get sent via communicate (so last change was actually useless), and readline still works fine apparently.

---


*Comments from [Reviewable](https://reviewable.io/reviews/google/mobly/456)*

**Comment by xpconanfan** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[mobly/controllers/android_device_lib/adb.py, line 208 at r1](https://reviewable.io/reviews/google/mobly/456#-LEVREEG3xFik2sES1m-:-LEVWe9O3f0Kl3BKhQw1:b-i1rbtx) ([raw file](https://github.com/google/mobly/blob/69c602b0ca226dfc73c16408380b696eb19896ca/mobly/controllers/android_device_lib/adb.py#L208)):*
<details><summary><i>Previously, winterfroststrom wrote…</i></summary><blockquote>

Yes.

Basically, if the process dies, then apparently stdout can still have buffered output, which doesn't get sent via communicate (so last change was actually useless), and readline still works fine apparently.
</blockquote></details>

Fascinating!
Can you add a comment here to preserve this learning :)

---


*Comments from [Reviewable](https://reviewable.io/reviews/google/mobly/456#-:-LEVWj_j98REjrtmePoW:bnfp4nl)*

**Comment by winterfroststrom** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[mobly/controllers/android_device_lib/adb.py, line 208 at r1](https://reviewable.io/reviews/google/mobly/456#-LEVREEG3xFik2sES1m-:-LEVY8cS4uIoypQMg792:b-896fix) ([raw file](https://github.com/google/mobly/blob/69c602b0ca226dfc73c16408380b696eb19896ca/mobly/controllers/android_device_lib/adb.py#L208)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Fascinating!
Can you add a comment here to preserve this learning :)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/google/mobly/456)*
